plugins/inputs/sqlserver/sqlserver.go

```diff
@@ -337,9 +352,18 @@ func (s *SQLServer) Stop() {
 	}
 }
 
-func (s *SQLServer) gatherServer(pool *sql.DB, query Query, acc telegraf.Accumulator) error {
+func (s *SQLServer) gatherServer(pool *sql.DB, server string, query Query, acc telegraf.Accumulator) error {
+	var rows *sql.Rows
+	var err error
+
 	// execute query
-	rows, err := pool.Query(query.Script)
+	if query.HasCachedData {
+		cachedData, _ := query.DataCache.Get(server)
+		rows, err = pool.Query(query.Script, sql.Named("p1", cachedData))
```

Comment:
what's the purpose of the cache here?

Reply:
@ssoroka, in the current plugin implementation the cache is used to save state of Query Store queries. More info about saving state in Query Store queries you can get from PR #8465 and from comment [https://github.com/influxdata/telegraf/pull/8465#issuecomment-751884278](https://github.com/influxdata/telegraf/pull/8465#issuecomment-751884278)

Reply:
@ssoroka can this be marked resolved?